module/zfs/dmu_objset.c

```diff
@@ -499,10 +524,10 @@ dmu_objset_open_impl(spa_t *spa, dsl_dataset_t *ds, blkptr_t *bp,
 	os->os_zil = zil_alloc(os, &os->os_zil_header);
 
 	for (i = 0; i < TXG_SIZE; i++) {
-		list_create(&os->os_dirty_dnodes[i], sizeof (dnode_t),
-		    offsetof(dnode_t, dn_dirty_link[i]));
-		list_create(&os->os_free_dnodes[i], sizeof (dnode_t),
-		    offsetof(dnode_t, dn_dirty_link[i]));
+		multilist_create(&os->os_dirty_dnodes[i], sizeof (dnode_t),
+		    offsetof(dnode_t, dn_dirty_link[i]),
+		    zfs_arc_num_sublists_per_state,
```

Comment:
Yes, this was mostly out of convenience.  In theory we might want it to be different, but in practice I would be surprised if that ever happened.  I don't mind making it a different variable if you want, but another option would be to rename/move the existing variable to be more obviously generic (e.g. `zfs_multilist_num_sublists`)